### PR TITLE
Viewbox implementation

### PR DESCRIPTION
**Pull request by donandren**
- What does the pull request do?
Add Viewbox and simplifies LayoutTransform control
- What is the current behavior?
we don't have Viewbox
- What is the updated/expected behavior with this PR?

Checklist:

- [x] Added unit tests (if possible)?
- [x] Added XML documentation to any related classes

### CODE REVIEW

**Review by grokys — CHANGES_REQUESTED** [review]
I created a [ViewBoxPage](https://gist.github.com/grokys/b11ff7e90257640c3f7b62807e0fc498) for the control catalog and compared it with the same [page in WPF](https://gist.github.com/grokys/8d88d9944f7cebee90c15d3bf5998bef). The results don't look quite right.

WPF looks as I'd expect it to:

image

However, Avalonia doesn't look right:

image

Not quite sure what's causing the difference. Is this expected?

**Comment by donandren** [comment]
updated default of ClipToBounds to true (which i missed initially) to behave as wpf.
I think differences in wpf vs. avalonia are not caused by Viewbox itself.
not sure where they are coming from, but

-  GridRowDefinitions with Auto height in a StackPanel definatelly is a complex scenario and could be misleading.
- having Path report it's default bounds might not be working as expected, not sure

anyway made tiny modifications to example page and it look to be it's working fine:
https://gist.github.com/donandren/3105ed68cdb4f7871abf15045ae671bf

File: src/Avalonia.Controls/ViewBox.cs [hunk 1]
```diff
@@ -0,0 +1,123 @@
+﻿using System;
+using Avalonia.Media;
+
+namespace Avalonia.Controls
+{
+    /// <summary>
+    /// Viewbox is used to scale single child.
+    /// </summary>
+    /// <seealso cref="Avalonia.Controls.Decorator" />
+    public class Viewbox : Decorator
```
[the remark this answers is not in the document]
Comment (donandren) [reply src/Avalonia.Controls/ViewBox.cs]: fixed

**Comment by grokys** [comment]
Yeah, I was wondering if the problem might be with `Grid`. Might be worth investigating this separately... Also could you add the `ViewboxPage` to ControlCatalog? Or I can add it if you'd prefer.

**Review by grokys — APPROVED** [review]
LGTM! Just pushed a quick fix to alphabetically order the pages in ControlCatalog, hope that's OK!

**Pull request merged**